lib/Dialect/HW/Transforms/CMakeLists.txt

```diff
@@ -20,4 +20,5 @@ add_circt_dialect_library(CIRCTHWTransforms
   MLIRIR
   MLIRPass
   MLIRTransformUtils
+  MLIRInferIntRangeCommon
```

Comment:
How come this is necessary? You don't seem to be touching anything else in `lib/Dialect/HW/Transforms`